pipeline/07-export.R

```diff
@@ -271,40 +271,71 @@ assessment_card <- dbGetQuery(
 #- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 message("Preparing data for Desk Review export")
 
-# Get the number of apartments for class 211 and 212 properties. Since
-# char_apts is recorded on the card level, we need to aggregate them to
-# the PIN level
-num_apts_by_pin <- assessment_card %>%
-  filter(char_class == "211" | char_class == "212") %>%
-  select(meta_pin, char_apts) %>%
-  # Convert the long format for char_apts to a shorter one that's easier
+format_char_apts <- function(char_apts) {
+  # Convert the long format for char_apts to a shorter format that's easier
   # to scan in a spreadsheet
-  mutate(
-    char_apts = case_when(
-      char_apts == "NONE" | is.na(char_apts) ~ NA_character_,
-      char_apts == "TWO" ~ "2",
-      char_apts == "THREE" ~ "3",
-      char_apts == "FOUR" ~ "4",
-      char_apts == "FIVE" ~ "5",
-      char_apts == "FIX" ~ "6",
+  return(
+    case_when(
+      is.na(char_apts) | tolower(char_apts) == "none" ~ NA_character_,
+      tolower(char_apts) == "two" ~ "2",
+      tolower(char_apts) == "three" ~ "3",
+      tolower(char_apts) == "four" ~ "4",
+      tolower(char_apts) == "five" ~ "5",
+      tolower(char_apts) == "six" ~ "6",
       TRUE ~ "Missing"
     )
-  ) %>%
-  # Adding the number of units might get confusing, since these classes are only
-  # supposed to have 2-6 units, so concatenate them as strings instead
-  summarize(char_apts = paste(char_apts, collapse = ", "), .by = "meta_pin")
+  )
+}
 
-num_commercial_units_by_pin <- assessment_card %>%
-  filter(char_class == "212") %>%
-  select(meta_pin, char_ncu) %>%
-  summarize(char_ncu = paste(char_ncu, collapse = ", "), .by = "meta_pin")
+summarize_char <- function(col) {
+  # Summary function that can take a column name `col` and return a
+  # comma-separated string of all the distinct, not-null values in that column.
+  # Note that nulls are preserved if and only if _all_ values in the column
+  # are null
```

Comment:
While multi-card parcels are not the norm, so the majority of parcels won't be affected, I'm worried that this aggregation behavior will be confusing to end users of the workbooks. I considered adding notes to the columns explaining why some cells contain comma-separated values but decided that it felt unlikely that a confused reader would check a column-level note to answer a question like that. Do you have ideas about how best to explain this change? Or am I overthinking it?

Reply:
I think you're overthinking it. Anyone working in res should know what a PIN can have multiple cards and therefore multiple characteristic values.